### PR TITLE
add `filterNot` to FunctorFilter

### LINKED ISSUE #3022: Add `filterNot` to FunctorFilter syntax
Just a general idea. If it sounds good, I'll implement it :)

### PR DESCRIPTION
**Pull request by gagandeepkalra**
resolves #3022

### CODE REVIEW

File: laws/src/main/scala/cats/laws/FunctorFilterLaws.scala [hunk 1]
```diff
@@ -24,6 +24,10 @@ trait FunctorFilterLaws[F[_]] {
   def filterConsistentWithMapFilter[A](fa: F[A], f: A => Boolean): IsEq[F[A]] =
     F.filter(fa)(f) <->
       F.mapFilter(fa)(a => if (f(a)) Some(a) else None)
+
+  def filterNotConsistentWithMapFilter[A](fa: F[A], f: A => Boolean): IsEq[F[A]] =
+    F.filterNot(fa)(f) <->
+      F.mapFilter(fa)(a => Some(a).filterNot(f))
```
Comment (LukaJCB) [inline laws/src/main/scala/cats/laws/FunctorFilterLaws.scala]: Why not instead make a `filterNotConsisentWithFilter` `filter(fa)(f) <-> filterNot(fa)(a => !f(a))`? :)

File: core/src/main/scala/cats/FunctorFilter.scala [hunk 2]
```diff
@@ -66,4 +66,11 @@ trait FunctorFilter[F[_]] extends Serializable {
    */
   def filter[A](fa: F[A])(f: A => Boolean): F[A] =
     mapFilter(fa)(a => if (f(a)) Some(a) else None)
+
+  /**
+   * Apply a filter to a structure such that the output structure contains all
+   * `A` elements in the input structure that do not satisfy the predicate `f`.
+   */
+  def filterNot[A](fa: F[A])(f: A => Boolean): F[A] =
+    filter(fa)(!f(_))
```
Comment (gagandeepkalra) [inline core/src/main/scala/cats/FunctorFilter.scala]: you're right @LukaJCB, this suits more to the documentation

**Comment by takayahilton** [comment]
Could you add `filterNot` to LazyList?

**Comment by gagandeepkalra** [comment]
Hello, is there something pending before we check this in?

**Comment by travisbrown** [comment]
Looks good to me!

**Pull request merged**